### PR TITLE
Fixes Ethernet support in latest builds

### PR DESCRIPTION
**Pull request by AdrianSoundy**
## Description

There was issues in the latest build not including the Ethernet support
- Ethernet option were incorrect
- ETHERNET_SUPPORT flag not being seen by other modules
- Connects not working as Wireless in enabled but not connected by default.
- DHCP issues when testing with sample Network Client

The only IDF config that supported Ethernet correctly was the sdkconfig.default used by LILYGO build. The binutils.ESP32.cmake looked for specific patterns in sdkconfig to update Ethernet configuration.

This Ethernet config has been updated to include option of SPI EThernet controllers in a build.
The changed sdkconfig ethernet section has been copied to all available IDF configs so all build type can have the option of Ethernet ( Internal or SPI )

The TCP connects not working was caused by having the Wireless interface enabled. To make this simpler for builds with Ethernet interface included it will now disable wireless on first start when config built and saved. 

The problems with DHCP were found to be caused by Network helper enabling DHCP at the same time DHCP was running in background. System.Net changed and updated to only Enable DHCP when not already enabled, which resolved problem.

As we now have option of supporting multiple PHY the NF_ESP32_Ethernet_Lan8720.cpp has been updated and renamed.  Support included for more internal PHY types and the options of SPI controllers like WIZ5500. 

These currently can only be enabled by changing options in esp32_ethernet_options.h.in but later we can enable it to be selected from build or dynamically from managed code. Out of scope for this PR.   I don't currently have hardware to test it on. Have wiz5500 on order.

## Motivation and Context
Had project that used Ethernet but didn't work with latest builds
also
- Fixes nanoFramework/Home#951

## How Has This Been Tested?

An updated version of sample Network client used. Sample has also be updated with latest Network Helper.

On the Olimex POE ISO board which has same config as LILYGO

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: targets/ESP32/_IDF/sdkconfig.debug_nopsram_ble.esp32 [hunk 1]
```diff
@@ -0,0 +1,1468 @@
+#
```
Comment (josesimoes) [inline targets/ESP32/_IDF/sdkconfig.debug_nopsram_ble.esp32:1]: I believe this sdkconfig wasn't meant to be in the PR... please confirm

File: targets/ESP32/_IDF/sdkconfig.default_pico [hunk 2]
```diff
@@ -343,8 +343,29 @@ CONFIG_ESP_MAC_ADDR_UNIVERSE_WIFI_AP=y
 
 #
 # Ethernet
+# 
+# Parameters configured in binutils.ESP32.CMAKE
 #
-CONFIG_ETH_ENABLED=n
+CONFIG_ETH_ENABLED=y
```
Comment (josesimoes) [inline targets/ESP32/_IDF/sdkconfig.default_pico:349]: Does it make sense to enable these for the PICO?

File: targets/ESP32/_Network/esp32_ethernet_options.h.in [hunk 3]
```diff
@@ -8,11 +8,47 @@
...
+// Default for Phy adr is 0
+#cmakedefine ETH_PHY_ADDR @ETH_PHY_ADDR@
+
+
+// Use internal ethernet 
+#define ESP32_ETHERNET_INTERNAL
+
+// Select one of following PHY for Internal Ethernet
+#define ESP32_ETHERNET_PHY_LAN8720
+//#define ESP32_ETHERNET_PHY_IP101
```
Comment (josesimoes) [inline targets/ESP32/_Network/esp32_ethernet_options.h.in:28]: I'm not very keen of this approach... this is a template file therefore it's not supposed to require any tweaks to build one target or the other. Not having to edit it to add/remove build options.

I get it that this is a convenient file to manage all these options. If that's the case, then let's expose the build options properly and map them properly. Let's follow up on Discord.

**Review by josesimoes — CHANGES_REQUESTED** [review]

File: targets/ESP32/_IDF/sdkconfig.default_pico (hunk 2 above)
Comment (AdrianSoundy) [reply targets/ESP32/_IDF/sdkconfig.default_pico:349]: I enabled the Ethernet drivers for all build types. It just means the driver will be available in Library if anybody enables it.  With the SPI controllers available that's likely to happen.  Having this option doesn't increase size of firmware just increases build time slightly.

File: targets/ESP32/_Network/esp32_ethernet_options.h.in (hunk 3 above)
Comment (AdrianSoundy) [reply targets/ESP32/_Network/esp32_ethernet_options.h.in:28]: This was only a temporary solution to allow testing until we add some proper build options. I wanted to handle that in separate PR.

File: targets/ESP32/_IDF/sdkconfig.debug_nopsram_ble.esp32 (hunk 1 above)
Comment (AdrianSoundy) [reply targets/ESP32/_IDF/sdkconfig.debug_nopsram_ble.esp32:1]: Currently the debug builds don't include debug symbols for debugging

File: targets/ESP32/_Network/esp32_ethernet_options.h.in (hunk 3 above)
Comment (josesimoes) [reply targets/ESP32/_Network/esp32_ethernet_options.h.in:28]: Understood. Let's have that in queue asap, so this doesn't be left hanging for too long.

**Review by josesimoes — APPROVED** [review]

**Comment by AdrianSoundy** [comment]
Not sure why you removed Bug label.  The Ethernet support is currently broken.

**Comment by josesimoes** [comment]
> Not sure why you removed Bug label. The Ethernet support is currently broken.

Because the root cause was in the build system. Not exactly in the code. 
OK there where changes required in the code, but still... 😁

**Pull request merged**